delta/plugins/project-deletion/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/projectdeletion/ProjectDeletionPlugin.scala

```diff
@@ -0,0 +1,164 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.projectdeletion
+
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.IOUtils
+import ch.epfl.bluebrain.nexus.delta.plugins.projectdeletion.model.ProjectDeletionConfig
+import ch.epfl.bluebrain.nexus.delta.sdk.ProjectResource
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.Pagination
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchParams.ProjectSearchParams
+import ch.epfl.bluebrain.nexus.delta.sdk.plugin.Plugin
+import ch.epfl.bluebrain.nexus.delta.sdk.projects.model.ProjectRejection
+import ch.epfl.bluebrain.nexus.delta.sdk.projects.{Projects, ProjectsStatistics}
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Identity
+import com.typesafe.scalalogging.Logger
+import fs2.Stream
+import monix.bio.{Fiber, IO, Task, UIO}
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Identity.Subject
+
+import java.time.{Duration, Instant}
+import scala.concurrent.duration.DurationInt
+
+/**
+  * ProjectDeletion process reference, also the plugin instance that allows for graceful stop. Once the instance is
+  * constructed, the process would be already running.
+  *
+  * @param fiber
+  *   the underlying process fiber which controls the process
+  */
+class ProjectDeletionPlugin private (fiber: Fiber[Throwable, Unit]) extends Plugin {
+
+  private val logger: Logger = Logger[ProjectDeletionPlugin]
+  logger.info("Project Deletion periodic check has started.")
+
+  override def stop(): Task[Unit] = fiber.cancel.timeout(10.seconds).flatMap {
+    case Some(()) => UIO.delay(logger.info("Project Deletion plugin stopped."))
+    case None     => UIO.delay(logger.error("Project Deletion plugin could not be stopped in 10 seconds."))
+  }
+}
+
+object ProjectDeletionPlugin {
+  implicit private class BooleanTaskOps(val left: Boolean) extends AnyVal {
+    def `<||>`(right: Task[Boolean]): Task[Boolean] = {
+      if (left) {
+        Task.pure(true)
+      } else {
+        right
+      }
+    }
+
+    def `<&&>`(right: Task[Boolean]): Task[Boolean] = {
+      if (left) {
+        right
+      } else {
+        Task.pure(false)
+      }
+    }
+  }
+
+  def projectDeletionPass(
+      allProjects: UIO[Seq[ProjectResource]],
+      deleteProject: ProjectResource => IO[ProjectRejection, Unit],
+      config: ProjectDeletionConfig,
+      lastEventTime: ProjectResource => Task[Instant]
```

Comment:
Same, this should be an UIO.
In the end, you should not have any Task in the conditions as none of them should fail as this would prevent to process other projects